fabric-api-base/src/main/java/net/fabricmc/fabric/api/event/EventFactory.java

```diff
@@ -75,6 +77,52 @@ public static void invalidate() {
 		return EventFactoryImpl.createArrayBacked(type, emptyInvoker, invokerFactory);
 	}
 
+	/**
+	 * Create a simple "array-backed" Event instance.
+	 *
+	 * @param eventT         The event class type.
+	 * @param listenerT      The listener class type.
+	 * @param make           Factory function for turning consuemrs into listeners
+	 * @param execute        Function for executing listeners
+	 * @param <T>            The event type.
+	 * @param <F>            The listener type.
+	 * @return The Event instance.
+	 */
+	public static <T, F> Event<F> createSimpleArrayBacked(Class<T> eventT, Class<F> listenerT,
```

Comment:
I don't fully think we need these - we can avoid the alloc of consumers or whatever and the extra invokedynamic for a few lines of code. Yes it's ugly but a longer term solution would generate these with asm.

Reply:
I originally created a `Consumer<T>`-like for my own convenience but because of the lax requirement on event inputs that created lots of boilerplate anyway. See the comment above. Also see #1271, since there wasn't a tracked issue for this yet. I will revert this for now.